services/wallet/saved_addresses.go

```diff
@@ -2,54 +2,214 @@ package wallet
 
 import (
 	"database/sql"
+	"fmt"
+	"time"
 
 	"github.com/ethereum/go-ethereum/common"
 )
 
+var SyncClockCreatedEditedHere uint64 = 0
+
+type SavedAddressMeta struct {
+	Removed     bool
+	SyncClock   uint64 // clock of the last sync
```

Comment:
What is `SyncClock` used for?
I can see being set etc, but I can't see any usage, do you know where it's used?

Reply:
That is part of the plan to do an incremental manual sync. I realized late that we don't store a sync clock for paired devices and I can't use it for now. However, I think will be great to have it for the future and send only new changes to paired devices based on the last sync timestamp. However, this requires changes that don't help the current feature. Should I remove it completely and bring it back later on?

Reply:
This is similar to other data types like bookmarks that have a delete_at and a clock

Reply:
It would be a bit more complex as you don't have only 2 devices, you might have 3, or you might remove one and add one etc, so you would have to keep track of the state of the sync for each device

We can revisit later I think, if necessary, for now it's best to remove it I think

Reply:
> This is similar to other data types like bookmarks that have a delete_at and clock

Ah that would be a different concern, not sure about bookmarks, also seems like it's not used, I'd avoid unnecessary fields unless there's an actual usage, since `clock` should signal already the last update and it's generally all we need in most cases

Reply:
> It would be a bit more complex as you don't have only 2 devices, you might have 3, or you might remove one and add one etc, so you would have to keep track of the state of the sync for each device
> 
I see we keep a Lamport time stamp that should be consistent between each device and might be what I was searching for. As you mentioned this requires investing some time and I'm not sure if the bandwidth saved brings worthy benefits at the present moment, will remove it for now.